concordium-consensus/src/Concordium/KonsensusV1/Types.hs

```diff
@@ -919,3 +974,118 @@ computeBlockHash bhh bqh =
 
 instance HashableTo BlockHash BakedBlock where
     getHash bb = computeBlockHash (getHash bb) (getHash bb)
+
+-- |A collection of signatures
+-- This is a map from 'FinalizerIndex' to the actual signature message.
```

Comment:
I'm just taking a wild guess as to what you will use this for: you want to be able to check if someone has signed two different blocks or two different timeouts in the same round. This data structure seems fine for that purpose. But if we want to be able to flag someone who is double signing, we need to also find the signature proving that they are double signing, and this data structure only stores the message, not the signature. So maybe we should add the signature as well.